dbt/adapters/snowflake/impl.py

```diff
@@ -259,3 +276,15 @@ def valid_incremental_strategies(self):
     def debug_query(self):
         """Override for DebugTask method"""
         self.execute("select 1 as id")
+
+    def capabilities(self) -> List[CapabilitySupport]:
```

Comment:
How would this be used? My gut is that the value of this is for `dbt-core` to inspect the adapter for a given capability, regardless of what's in the adapter. That would mean that this needs to be the full enum, not just what's in the adapter.

Reply:
I believe the situation is clarified with the new design, where the failure of an adapter to declare support results in the state of support simply being "Unknown" from core's perspective.